firmware-support/bittide-sys/src/gather_unit.rs

```diff
@@ -1,82 +1,33 @@
 // SPDX-FileCopyrightText: 2022 Google LLC
 //
 // SPDX-License-Identifier: Apache-2.0
-
-use fdt::node::FdtNode;
-
-use crate::{utils::matches_fdt_name, ComponentLoadError};
-
-pub struct GatherUnit<const FRAME_SIZE: usize> {
+pub struct GatherUnit<const MEM_SIZE: usize> {
     memory: *mut u8,
     metacycle_register: *const u8,
```

Comment:
I would prefer instead to have just the base address stored and create an associated constant for any other registers for the peripheral, like is done for `ClockControl` (`firmware-support/bittide-sys/src/clock_control.rs`) and `Clock` (`firmware-support/bittide-sys/src/time.rs`). Should be a free memory footprint decrease with no performance hit (or sometimes even a notable performance increase).